s3_bucket_example.tf

```diff
@@ -1,5 +1,6 @@
 resource "aws_s3_bucket" "example_x" {
-  bucket        = "Example Bucket for use by Service X"
+  count         = "${replace(replace(var.bucket_name,"/.+/","1"),"/^$/","0")}"
```

Comment:
I get it. My goal here was just to make testing possible with no need to create a bucket as bucket name must be unique but at the same time leaving the functionality in place. So actually I made bucket_name param optional, that's it. If you set it to sth, you'll get a bucket. If you don't, you won't have one.